pages/1.11/security/ent/secrets/secrets-api/index.md

```diff
@@ -94,11 +97,42 @@ curl -H "Authorization: token=$(dcos config show core.dcos_acs_token)"
 Authentication tokens expire after five days by default. If your program needs to run longer than five days, you will need a service account. Please see [Provisioning custom services](/1.11/security/ent/service-auth/custom-service-auth/) for more information.
 
 
+## Text and binary secrets
+
+From DC/OS Enterprise version 1.11 the Secrets API supports text and binary secrets. A text secret is a text value encoded with UTF-8 encoding. The Secrets API passes through this value to a secret store backend without any modification. A binary secret is automatically encoded with the base64 codec to a text format before it is passed to a secrets store backend.
+
+Use cases for secret types:
+
+* **Text**: can support storing text passwords, JSON documents, YAML documents
+
+* **Binary**: all the text secrets and also binary files, like Java KeyStore, Kerberos keytab files
+
+**Note**: It is not recommended to use binary secrets values as environment variables. POSIX environment variables cannot contain `NUL` bytes.
+
+### Storing a secret
+
+When an HTTP client is sending a secret value to the Secrets API it should send a `Content-Type` header containing information about a secret type. For backward compatibility with existing HTTP clients if a `PUT`/`PATCH` request is sent without a `Content-Type` header the DC/OS Secrets service assumes that the secret is a text secret and it behaves as if `application/json` was included in a request.
+
+Endpoints which accept both `application/json` and `application/octet-stream` values:
+
+* `PUT /secret/{store}/{path/to/secret}`
+* `PATCH /secret/{store}/{path/to/secret}`
+
+**Note**: The maximum file size for a `BASE64` encoded file is approximately one MiB, subtracting approximately one KB for the secret store metadata. This gives a room for storing a file of approximately maximum size of 768 KiB.
+
+### Retrieving a secret
+
+To retrieve a secret using the Secrets API an HTTP client issues a `GET /secret/{store}/{path/to/secret}` HTTP request. A request can contain an `Accept` header to indicate the Secrets API in what format would client like to get a secret value. It is possible to use `Accept: application/json` or `Accept: application/octet-stream` headers. If a client does not provide an `Accept` header the Secrets API returns a secret value in the format in which it was stored in the DC/OS Secrets service. The Secrets API sends a `Content-Type` header in an HTTP response to indicate what type of value is being sent back in the response.
```

Comment:
> to indicate the Secrets API

"to indicate *to* the..."

Reply:
> in what format would client like to get a secret value

"in what format the client would like to get a secret value" or maybe "the secret value", and maybe not "would like" but "should receive" or something more formal.